src/app/core/services/notifications/notification-wrapper.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core'
+
+import { DefaultNotificationType } from '../../../../assets/data/defaultConfig'
+import { ConfigKeys } from '../../../shared/enums/config'
+import { NotificationMessagingType } from '../../../shared/models/notification-handler'
+import { RemoteConfigService } from '../config/remote-config.service'
+import { FcmRestNotificationService } from './fcm-rest-notification.service'
+import { FcmXmppNotificationService } from './fcm-xmpp-notification.service'
+import { LocalNotificationService } from './local-notification.service'
+
+@Injectable()
+export class NotificationWrapperService {
```

Comment:
here you are using what is called a `Decorator` design pattern. This class needs to extend `NotificationService` too.

Reply:
Please see my comment on providing the NotificationService below. This wrapper will not be needed then